router.go

```diff
@@ -54,6 +56,11 @@ func (route *Route) addHandlerTo(routeStr string, h *handler) {
 				matchRoute.PathRouteHandler = make(map[string]*handler)
 			}
 			if _, ok := matchRoute.PathRouteHandler[h.method]; !ok {
+				if matchRoute.optionsHandler == nil {
+					matchRoute.optionsHandler = newOptionsHandler()
+				}
+				matchRoute.optionsHandler.addMethod(h.method)
+				matchRoute.PathRouteHandler["OPTIONS"] = matchRoute.optionsHandler.build()
 				matchRoute.OwnHandler = true
 				matchRoute.PathRouteHandler[h.method] = h
```

Comment:
I think we should use `matchRoute.AddHandler(h)` to setup handler and add methods to options handler in `AddHandler` function.